### PR TITLE
PLA-6250 add new design workflow routes to documentation

### PR DESCRIPTION
**Pull request by talltea**
# Citrine Python PR

## Description 
Adds documentation for the new Design Workflow routes. Does not yet remove the documentation for the new routes.

### PR Type:
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

### Adherence to team decisions
- [ ] I have added tests for 100% coverage
- [ ] I have written Numpy-style docstrings for every method and class.
- [ ] I have communicated the downstream consequences of the PR to others.
- [ ] I have bumped the version in setup.py

### CODE REVIEW

File: docs/source/workflows/design_workflows.rst [hunk 1]
```diff
@@ -79,41 +107,90 @@ The following demonstrates how to trigger workflow execution, wait for the desig
...
+    best_score = scores[0]
+    print(best_score)
+
+    # we can confirm the best candidate is at the head of the list using
+    # this candidate will be the same as best_candidate above
+    candidate_with_max_shear_modulus = max(candidates, key=lambda candidate: float(candidate['Shear modulus']))
+    print(candidate_with_max_shear_modulus)
+
+
+In the updated design execution workflow, results are paginated and returned as `DesignCandidate` objects.
```
Comment (maxhutch) [inline docs/source/workflows/design_workflows.rst:150]:
```suggestion
In the updated design execution workflow, results are paginated and returned as :class:`~citrine.informatics.design_candidate.DesignCandidate` objects.
```
Add a link to the docs.

**Review by maxhutch — COMMENTED** [review]
I think it'd be useful to describe `DesignMaterial` and `DesignVariable` a little more.  It doesn't have to be a book, but explaining that design candidates contain DesignMaterials, which are simpler approximations ("projections") of the materials information about a design candidate to help users find the candidates that they want, and how each of the DesignVariables approximates the distribution of values the variable might take.

**Comment by talltea** [comment]
> I think it'd be useful to describe `DesignMaterial` and `DesignVariable` a little more. It doesn't have to be a book, but explaining that design candidates contain DesignMaterials, which are simpler approximations ("projections") of the materials information about a design candidate to help users find the candidates that they want, and how each of the DesignVariables approximates the distribution of values the variable might take.

I've added a section

**Review by maxhutch — APPROVED** [review]
LGTM!

**Pull request merged**